src/WaveTrack.cpp

```diff
@@ -73,8 +73,10 @@ bool AreAligned(const WaveClipPointers& a, const WaveClipPointers& b)
       return false;
 
    const auto compare = [](const WaveClip* a, const WaveClip* b) {
-      return a->GetStartTime() == b->GetStartTime() &&
-         a->GetNumSamples() == b->GetNumSamples();
+      return a->GetPlayStartTime() == b->GetPlayStartTime() &&
+         a->GetSequenceStartTime() == b->GetSequenceStartTime() &&
+         a->GetPlayEndTime() == b->GetPlayEndTime() &&
+         a->GetSequenceEndTime() == b->GetSequenceEndTime();
```

Comment:
Buried in all these changes is one very interesting thing here:  the need for a more strict definition of correspondence of clips.

A detail like this deserves comments, both in the code here, and also in the git commit comments.

I'd like to encourage you to write more comments, of both kinds, about such changes of behavior.

Reply:
Updated both commit message and function comment

Reply:
Thanks for that